src/tool/xmeta/cpp/ast_to_st_listener.cpp

```diff
@@ -0,0 +1,206 @@
+#include <algorithm>
+#include <stdexcept>
+#include "symbol_table.h"
+#include "models/xmeta_models.h"
+#include "ast_to_st_listener.h"
+
+using namespace xlang::xmeta;
+extern bool semantic_error_exists;
+
+std::map<std::string, enum_semantics> enum_semantics_map
+{
+    { "Int8", enum_semantics::Int8 },
+    { "UInt8", enum_semantics::Uint8 },
+    { "Int16", enum_semantics::Int16 },
+    { "UInt16", enum_semantics::Uint16 },
+    { "Int32", enum_semantics::Int32 },
+    { "UInt32", enum_semantics::Uint32 },
+    { "Int64", enum_semantics::Int64 },
+    { "UInt64", enum_semantics::Uint64 }
+};
+
+ast_to_st_listener::ast_to_st_listener(xmeta_symbol_table & symbol_table) :
+    m_st{ symbol_table }
+{ }
+
+bool ast_to_st_listener::extract_enum_member(XlangParser::Enum_member_declarationContext *ast_enum_member, std::shared_ptr<enum_model> const& new_enum)
```

Comment:
The semantics of this return value are unclear - is an error indicated by `true` or `false`? If you really want to accomplish this via a return value instead of throwing, then that's where `bool` enums come in handy to make the return type and values communicate meaning.
http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Ri-typed